app/src/main/java/com/jshvarts/todoapp/notelist/ui/NoteListUiContract.kt

```diff
@@ -1,30 +1,44 @@
 package com.jshvarts.todoapp.notelist.ui
 
 import android.os.Parcelable
-import com.jshvarts.todoapp.arch.UiAction
 import com.jshvarts.todoapp.arch.UiEffect
 import com.jshvarts.todoapp.arch.UiState
 import com.jshvarts.todoapp.data.Note
 import kotlinx.parcelize.Parcelize
 import kotlinx.parcelize.RawValue
 
-sealed interface NoteListUiAction : UiAction {
-  object LoadList : NoteListUiAction
+@Parcelize
+sealed class NoteListPendingTodosUiState : UiState, Parcelable {
+  object Loading : NoteListPendingTodosUiState()
+
+  data class Success(
+    val data: @RawValue List<Note>
+  ) : NoteListPendingTodosUiState()
+
+  data class Error(
+    val throwable: Throwable? = null
+  ) : NoteListPendingTodosUiState()
 }
 
 @Parcelize
-sealed class NoteListUiState : UiState, Parcelable {
-  object Loading : NoteListUiState()
+sealed class NoteListCompletedTodosUiState : UiState, Parcelable {
```

Comment:
This looks the same as `NoteListPendingTodosUiState`. Can we rename `NoteListPendingTodosUiState` to something more general and reuse that?

Reply:
👍

Reply:
Done